### ISSUE TITLE
Column title bug

### ISSUE THREAD

**Issue by c349liu** [metabase/metabase#12739]
**Describe the bug**
When a table is joined multiple times in a question, columns resulted from different joins of that table cannot be consistently named differently. In other words, changing the title for one of the columns will result in other columns having the same title. 

**Logs**
Don't think you need this.

**To Reproduce**
1. Have two tables present in Metabase.
Table 1: user_id, field_1
Table 2: user_id, field_2
2. Create a custom question, select Table 1 as starting data
3. Select join data (left outer join). Select Table 2 where Table 1.user_id = Table 2.user_id. Do this multiple times.
4. Click Hide Editor
5. Click on one of the column titles named "Table 2 -> field 2". Select Formatting.
6. Rename the column title and click Done.
7. Now you should see that all "Table 2 -> field 2" columns have been renamed, instead of changing only one of them. 
**Expected behavior**
See step 7

**Screenshots**
Due to sensitivity of data, I will not share the screenshot. 

**Severity**
BI users are blocked by this bug. 

**Additional context**
One use case for the bug scenario is that for a table with a schema of (user_id::text, question::text, answer::text), we can filter by condition question = 'q3' and the answer field will be the users' answer for q3. When we are constructing a portfolio for a particular user, we might need to gather the answers from different questions for that user. Obviously, answers for different questions will be presented as individual columns and they should have different names. With the described bug in Metabase, we cannot accomplish this through the interface. 

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.106 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.5+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.5",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.5+10",
    "os.name": "Linux",
    "os.version": "4.15.0-1052-azure",
    "user.language": "en",
    "user.timezone": "UTC"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "10.11"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-04-10",
      "tag": "v0.35.2",
      "branch": "release-0.35.x",
      "hash": "f3b2857"
    },
    "settings": {
      "report-timezone": "Canada/Eastern"
    }
  }
}
```

**Comment by flamber** [comment]
Duplicate of #12685
